analysis/import/SCMLogParserV2/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparserv2/converter/ProjectConverter.kt

```diff
@@ -0,0 +1,78 @@
+package de.maibornwolff.codecharta.importer.scmlogparserv2.converter
+
+import de.maibornwolff.codecharta.importer.scmlogparserv2.input.VersionControlledFile
+import de.maibornwolff.codecharta.importer.scmlogparserv2.input.metrics.MetricsFactory
+import de.maibornwolff.codecharta.importer.scmlogparserv2.parser.VersionControlledFilesInGitProject
+import de.maibornwolff.codecharta.importer.scmlogparserv2.parser.VersionControlledFilesList
+import de.maibornwolff.codecharta.model.Edge
+import de.maibornwolff.codecharta.model.MutableNode
+import de.maibornwolff.codecharta.model.NodeType
+import de.maibornwolff.codecharta.model.PathFactory
+import de.maibornwolff.codecharta.model.Project
+import de.maibornwolff.codecharta.model.ProjectBuilder
+
+/**
+ * creates Projects from List of VersionControlledFiles
+ */
+class ProjectConverter(private val containsAuthors: Boolean) {
+
+    private val ROOT_PREFIX = "/root/"
+
+    private fun addVersionControlledFile(projectBuilder: ProjectBuilder, versionControlledFile: VersionControlledFile) {
+        val attributes = extractAttributes(versionControlledFile)
+        val edges = versionControlledFile.getEdgeList()
+        val fileName = versionControlledFile.filename.substringAfterLast(PATH_SEPARATOR)
+        val newNode = MutableNode(fileName, NodeType.File, attributes, "", mutableSetOf())
+        val path = PathFactory.fromFileSystemPath(
+            versionControlledFile.filename.substringBeforeLast(PATH_SEPARATOR, "")
+        )
+
+        projectBuilder.insertByPath(path, newNode)
+        edges.forEach { projectBuilder.insertEdge(addRootToEdgePaths(it)) }
+        versionControlledFile.removeMetricsToFreeMemory()
```

Comment:
Add Todo: Improve memory uitilization - inject metrics for calculations instead of creating a hard reference in VersionControlledFile.